Yank/multistate/multistatesampler.py

```diff
@@ -1140,14 +1188,18 @@ def _compute_energies(self):
     def _compute_replica_energies(self, replica_id):
         """Compute the energy for the replica in every ThermodynamicState."""
         # Initialize replica energies for each thermodynamic state.
-        energy_thermodynamic_states = np.zeros(self.n_states)
+        energy_thermodynamic_states = np.ma.zeros(self.n_states)
         energy_unsampled_states = np.zeros(len(self._unsampled_states))
 
         # Retrieve sampler state associated to this replica.
         sampler_state = self._sampler_states[replica_id]
 
+        # Determine neighborhood
+        state_index = self._replica_thermodynamic_states[replica_id]
+        neighborhood = self._neighborhood(state_index)
+
         # Compute energy for all thermodynamic states.
-        for energies, states in [(energy_thermodynamic_states, self._thermodynamic_states),
+        for energies, states in [(energy_thermodynamic_states, self._thermodynamic_states[neighborhood]),
```

Comment:
Don't you also need to slice the `energy_thermodynamic_states` here? Otherwise wont the line `for energy_idx, state_idx in enumerate(state_indices)` ([absolute reference to line](https://github.com/choderalab/yank/blob/18929cc0c66d498a131d3190e20f1f252e98cfcb/Yank/multistate/multistatesampler.py?utf8=%E2%9C%93#L1219)) fail, or at least assign energies to the wrong place?

Reply:
Good catch!